src/libraries/System.Private.CoreLib/src/System/Text/Latin1Utility.cs

```diff
@@ -74,7 +57,7 @@ private static unsafe nuint GetIndexOfFirstNonLatin1Char_Default(char* pBuffer,
                     // data or we approach the end of the buffer. It's possible we'll reread data; this is ok.
 
                     char* pFinalVectorReadPos = pBuffer + bufferLength - SizeOfVectorInChars;
-                    pBuffer = (char*)(((nuint)pBuffer + SizeOfVectorInBytes) & ~(nuint)(SizeOfVectorInBytes - 1));
+                    pBuffer = (char*)(((nuint)pBuffer + SizeOfVectorInBytes) & ~(SizeOfVectorInBytes - 1));
```

Comment:
Need to keep the `(nuint)` cast here. Otherwise the `a & ~b` results in `~b` being calculated before zero-extension, which could result in the high 32 bits of the address being dropped.

Reply:
Ditto